www/js/app.js

```diff
@@ -819,47 +821,45 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
             // Create (or replace) the "base" tag with our base URL
             $('#articleContent').contents().find('head').find("base").detach();
             $('#articleContent').contents().find('head').append("<base href='" + baseUrl + "'>");
+            
+            var currentProtocol = location.protocol;
+            var currentHost = location.host;
 
             // Convert links into javascript calls
             $('#articleContent').contents().find('body').find('a').each(function() {
+                var href = $(this).attr("href");
                 // Compute current link's url (with its namespace), if applicable
-                var url = regexpZIMUrlWithNamespace.test(this.href) ? this.href.match(regexpZIMUrlWithNamespace)[1] : $(this).attr("href");
-                if (url === null || url === undefined) {
-                    return;
+                var zimUrl = regexpZIMUrlWithNamespace.test(this.href) ? this.href.match(regexpZIMUrlWithNamespace)[1] : "";
+                if (href === null || href === undefined) {
+                    // No href attribute
                 }
-                var lowerCaseUrl = url.toLowerCase();
-                var cssClass = $(this).attr("class");
-
-                if (cssClass === "new") {
-                    // It's a link to a missing article : display a message
+                else if (href.length === 0) {
+                    // It's a link with an empty href, pointing to the current page.
+                    // Because of the base tag, we need to modify it
                     $(this).on('click', function(e) {
-                        alert("Missing article in Wikipedia");
-                        return false;
+                       return false; 
                     });
                 }
-                else if (url.slice(0, 1) === "#") {
-                    // It's an anchor link : do nothing
-                }
-                else if (url.substring(0, 4) === "http") {
-                    // It's an external link : open in a new tab
-                    $(this).attr("target", "_blank");
+                else if (regexpLocalAnchorHref.test(href)) {
+                    // It's an anchor link : we need to make it work with javascript
+                    // because of the base tag
+                    var anchor = href;
+                    $(this).on('click', function(e) {
+                        $('#articleContent').first()[0].contentWindow.location.hash = anchor;
+                        return false;
```

Comment:
In the above, is it necessary to define anchor, when its value is equal to href, so you could just use href? I tested commenting out `var anchor = href;` and just using `= href;` instead of `= anchor;` in the function, and it works fine (I tested it just in case there was a conflicting assignment of href at the time that the .onclick function runs). However I can see that it might be desirable to keep this for reasons of clarity in understanding the code, and if so that's fine.

Reply:
I agree, this variable is not necessary